packages/sdk/src/coingecko/coingecko.ts

```diff
@@ -68,7 +68,7 @@ class Coingecko {
     // annoying, but have to type this to iterate over entries
     type Result = {
       [address: string]: {
-        usd: number;
+        [currency: string]: number; // usd, eth, ...
```

Comment:
Not totally confident that this is the right adjustment. Basically, the field that we get back in the Result should be the same as the input currency string that we received from the caller.

Reply:
Yeah, I think this is the right way to go. You could try to build a generic function that would infer that the key is a literal type that was passed in, but simple things like that get overly annoying and complex in typescript :/